pkg/controllers/gacha-controller.go

```diff
@@ -100,23 +85,112 @@ func CreateUser(w http.ResponseWriter, r *http.Request) {
 // UpdateUser does not show any reponse.
 
 func UpdateUser(w http.ResponseWriter, r *http.Request) {
-	var updateUser = &models.User{}
-	utils.ParseBody(r, updateUser)
-	vars := mux.Vars(r)
-	userId := vars["userId"]
-	ID, err := strconv.ParseInt(userId, 0, 0)
-	if err != nil {
-		fmt.Println("error while parsing")
+	var updateUser models.User
+	utils.ParseBody(r, &updateUser)
+
+	// Extract the X-Token value from the request headers
+	token := r.Header.Get("X-Token")
+
+	// Retrieve the user based on the token
+	user := models.GetUserByToken(token)
+	fmt.Print(user)
+
+	if user == nil {
+		// User not found, return an appropriate error response
+		w.WriteHeader(http.StatusNotFound)
+		return
 	}
-	userDetails, db := models.GetUserById(ID)
+
+	// Update the user details if the fields are not empty
 	if updateUser.Name != "" {
-		userDetails.Name = updateUser.Name
+		user[0].Name = updateUser.Name
 	}
 	if updateUser.Token != "" {
-		userDetails.Token = updateUser.Token
+		user[0].Token = updateUser.Token
+	}
+
+	// Save the updated user details
+	err := models.UpdateUser(&user[0])
+	if err != nil {
+		// Error occurred while updating user, return an appropriate error response
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+}
+
+func ListCharacters(w http.ResponseWriter, r *http.Request) {
+
+	characters := models.GetAllCharacters()
+
+	res, _ := json.Marshal(characters)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(res)
+}
+
+func HandleGachaDraw(w http.ResponseWriter, r *http.Request) {
+
+	var reqBody models.GachaDrawRequest
+	err := json.NewDecoder(r.Body).Decode(&reqBody)
+	if err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+	fmt.Printf("Received request: %+v\n", reqBody)
+	characters := models.GetAllCharacters()
+	characterPool := generatecharacterPool(characters)
+	response := models.GachaDrawResponse{
+		Results: []models.CharacterResponse{},
 	}
-	db.Save(&userDetails)
+	// fmt.Println(reqBody.NumTrials)
+	for i := 0; i < reqBody.Times; i++ {
+		character := models.DrawCharacter(characters, characterPool) // Simulate drawing a character
+		fmt.Println(character)
+		response.Results = append(response.Results, models.CharacterResponse{
+			CharacterID: fmt.Sprintf("Character-%d", character.ID),
+			Name:        character.Name,
+		})
+
+	}
+
+	respBody, err := json.Marshal(response)
+	if err != nil {
+		http.Error(w, "Error creating response", http.StatusInternalServerError)
+		return
+	}
+
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
+	w.Write(respBody)
+
+}
+func generatecharacterPool(characters []models.Character) []models.Character {
```

Comment:
C should be capitalized.

Reply:
@GuiltyMorishita I have updated it in the latest commit.